### ISSUE TITLE
may be incorrect loss funtion?

### ISSUE THREAD

**Issue by JohnTargaryen** [SunnerLi/DeepFuse.pytorch#2]
hey there, I notice that your code for loss function in ssim_loss_function.py is quite similar to the code I found in https://stackoverflow.com/questions/39051451/ssim-ms-ssim-for-tensorflow

However, this seems to be SSIM LOSS function instead of MEF SSIM LOSS function. In
K. Ma, K. Zeng, and Z. Wang. Perceptual quality assessment for multi-exposure image fusion. IEEE Transactions on Image Processing, 24(11):3345–3356, 2015. , the author mentioned that "Direct use of the SSIM algorithm [27], however, is impossible, which requires a single perfect quality reference image. "

The paper Deepfuse's work is exactly on MEF(multi-exposure image fusion), so I would think there should be differences between SSIM LOSS and MEF SSIM LOSS, in other words, the loss function in your code may be incorrect?

Look forward to your reply, thx

**Comment by Henry-GongZY** [comment]
> hey there, I notice that your code for loss function in ssim_loss_function.py is quite similar to the code I found in https://stackoverflow.com/questions/39051451/ssim-ms-ssim-for-tensorflow
> 
> However, this seems to be SSIM LOSS function instead of MEF SSIM LOSS function. In K. Ma, K. Zeng, and Z. Wang. Perceptual quality assessment for multi-exposure image fusion. IEEE Transactions on Image Processing, 24(11):3345–3356, 2015. , the author mentioned that "Direct use of the SSIM algorithm [27], however, is impossible, which requires a single perfect quality reference image. "
> 
> The paper Deepfuse's work is exactly on MEF(multi-exposure image fusion), so I would think there should be differences between SSIM LOSS and MEF SSIM LOSS, in other words, the loss function in your code may be incorrect?
> 
> Look forward to your reply, thx

I totally agree with you, MEF-SSIM is quite different from SSIM indeed.
